### ISSUE TITLE
Api and watcher pods are not created on Kubernetes v1.25

### ISSUE THREAD

**Issue by enarha** [tektoncd/results#271]
# Expected Behavior
Pods start.

# Actual Behavior
Pods are not created. Error messages:
```
Warning  FailedCreate  44s (x7 over 3m25s)  replicaset-controller  (combined from similar events): Error creating: pods "tekton-results-api-b94c64d8c-m7mh8" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "tekton-results-api" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "tekton-results-api" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "tekton-results-api" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "tekton-results-api" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```
```
 Warning  FailedCreate  12m   replicaset-controller  Error creating: pods "tekton-results-watcher-785b577d86-wt9dc" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "tekton-results-watcher" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "tekton-results-watcher" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "tekton-results-watcher" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "tekton-results-watcher" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

# Steps to Reproduce the Problem

1. `kind create cluster --name tekton-results`. By default version v1.25.3 is deployed.
2. Follow the install instructions https://github.com/tektoncd/results/blob/main/docs/install.md
3. `kubectl get pods -n tekton-pipelines` does not show the api and watcher pods.

# Additional Info

- Kubernetes version:

  ```
Client Version: v1.25.3
Kustomize Version: v4.5.7
Server Version: v1.25.3
  ```

- Tekton Pipeline version:
v0.42.0

This seems to be related: https://kubernetes.io/blog/2022/08/25/pod-security-admission-stable/

**Comment by enarha** [comment]
Tested with the latest Kubernetes v.1.23 version (v1.23.13) with the same result.

**Comment by khrm** [comment]
I am able to create on 1.24.
Are you installing from release.yaml?

**Comment by enarha** [comment]
@khrm Yes. I follow strictly the install doc. In both cases I used `kind` to create my cluster.  I use `kind create cluster --name tekton-results --image kindest/node:v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61` to create the v1.23 cluster and skipping that image parameter uses the latest one (v1.25.3).

**Comment by enarha** [comment]
I was curious why the above issues appear when deploying using the release.yaml and does not appear when deploying using the script `test/e2e/01-install.sh`. I found out that the deployment manifests generated by kustomize in the second case include the security context https://github.com/tektoncd/results/blob/main/config/api.yaml#L61-L70 and https://github.com/tektoncd/results/blob/main/config/watcher.yaml#L67-L76 missing from the release.yaml. Unless they are omitted on purpose, I'll try to create a fix for the release.yaml.

**Comment by vdemeester** [comment]
@enarha note that the latest release of `results` is quite old. We need to do a `0.5.0` release that would have those in. See https://github.com/tektoncd/results/issues/237

**Comment by enarha** [comment]
@vdemeester Thanks for pointing that out. I was testing with `0.4.0`, but I see the change which added the security context is quite recent. Anyway, I can close this issue once `0.5.0` is out and focus on the issue blocking that release.

**Comment by adambkaplan** [comment]
I believe this has been fixed with the v0.5.0 release - it is able to run on Kubernetes 1.25 and above.
